app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java

```diff
@@ -17,8 +17,12 @@ public ParkingPresenter(ParkingModel newParkingModel, ParkingView newParkingView
 
     public void onParkingSizeCreationButtonPressed() {
         try {
-            parkingModel.setParkingSize(parkingView.getSizeSubmitted());
-            parkingView.showParkingSize(parkingModel.getParkingSize());
+            if ((Integer.parseInt(parkingView.getSizeSubmitted())) == 0) {
```

Comment:
Here you are parsing parking size twice, once in ParkingModel.setParkingSize and another time here. You should better move this check to ParkingModel and throw an IllegalArgumentException when size is equal or less than 0. Then you can catch here, just as NumberFormatException.

Reply:
Added IllegalArgumentException